pkg/embed/generator.go

```diff
@@ -0,0 +1,138 @@
+// +build ignore
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"go/format"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+	"text/template"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/download"
+
+	"github.com/code-ready/crc/pkg/crc/machine/hyperkit"
+	"github.com/code-ready/crc/pkg/crc/machine/libvirt"
+)
+
+const (
+	blobFileName string = "blob"
+)
+
+// Define vars for build template
+var conv = map[string]interface{}{"conv": fmtByteSlice}
+var tmpl = template.Must(template.New("").Funcs(conv).Parse(`package embed
+
+// Code generated by go generate; DO NOT EDIT.
+
+func init() {
+    {{- range $name, $file := . }}
+        Add("{{ $name }}", []byte{ {{ conv $file }} })
+    {{- end }}
+}`),
+)
+
+func fmtByteSlice(s []byte) string {
+	builder := strings.Builder{}
+
+	for _, v := range s {
+		builder.WriteString(fmt.Sprintf("%d,", int(v)))
+	}
+
+	return builder.String()
+}
+
+func main() {
+	embedFolder, err := ioutil.TempDir("", "crc-data")
+	if err != nil {
+		log.Fatal("Error creating temp directory:", err)
+	}
+	defer os.RemoveAll(embedFolder)
+
+	// Create embed folder
+	downloadedFiles, err := downloadDataFiles(os.Getenv("GOS"), embedFolder)
+
+	// Create map for filenames
+	configs := make(map[string][]byte)
+
+	// Adding files to embeded struct
+	for _, f := range downloadedFiles {
+		fileName := filepath.Base(f)
+		log.Printf("Packing %s in...\n", fileName)
+
+		b, err := ioutil.ReadFile(f)
+		if err != nil {
+			log.Fatalf("Error reading %s: %s", f, err)
+		}
+		// Add file name to map
+		configs[fileName] = b
+	}
+
+	if err != nil {
+		log.Fatal("Error walking through embed directory:", err)
+	}
+
+	generatedFileName := fmt.Sprintf("%s_%s.go", blobFileName, os.Getenv("GOS"))
```

Comment:
`GOS` or `GOOS`?

Reply:
Ok, I saw the updated commit log. Let's use a name for this var which does not look like a typo rather than resorting to documentation. `TARGET_OS`?